Classes/EventListener/AbstractImageOptimizer.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netlogix\Nximageoptimizer\EventListener;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use TYPO3\CMS\Core\Configuration\ExtensionConfiguration;
+use TYPO3\CMS\Core\Resource\AbstractFile;
+use TYPO3\CMS\Core\Resource\ProcessedFile;
+use TYPO3\CMS\Core\SingletonInterface;
+use TYPO3\CMS\Core\Utility\CommandUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+abstract class AbstractImageOptimizer implements SingletonInterface, LoggerAwareInterface
+{
+	use LoggerAwareTrait;
+
+	protected array $configuration;
+
+	public function __construct()
+	{
+		$this->configuration = GeneralUtility::makeInstance(ExtensionConfiguration::class)->get('nximageoptimizer');
+	}
+
+	protected function isEnabled(ProcessedFile $processedFile): bool
+	{
+		// this is needed for backwards compatibility
+		// @extensionScannerIgnoreLine
+		if (!(TYPO3_REQUESTTYPE & TYPO3_REQUESTTYPE_FE)) {
+			// this is not needed for TYPO3 backend and would break deferred image processing
+			return false;
+		}
+
+		if (get_class($this) == WebpCreator::class && $this->configuration['disableAutomaticWebpCreation']) {
```

Comment:
Can you move this check in the WebpCreator.

Reply:
I have reverted the respective commit